### ISSUE TITLE
Further tasks in the deep learning notebook.

### ISSUE THREAD

**Issue by samagra14** [aimacode/aima-java#464]
## High priority task:
1. Integrate charting library and incorporate charts in the notebook.
2. Reduce unnecessary imports.(Use * wherever possible). Do this immediately.
## Next milestones
1. Running resnets and inception net using tensorflow java api.
2. Add some descriptions of what saturating and non-saturating functions are.
3. Add training curves of the models trained.
4. Comparison of Adagrad vs gradient descent.
5. A naive implementation of gradient descent(using  ND4J).
6. A naive implementation of backprop(using ND4J).
7. Backprop vs Adagrad
8. Variations of training with depth(Architecture considerations)
9. Weight decay and dropout.
10. Transfer learning demo using imagenet pretrained features for another use.

Following this, we will focus on:
1. probabilistic PCA
2. Variational autoencoders.
3. Generative models.(Maybe)

**Comment by samagra14** [comment]
@jsuyash1514 have a look.
